gusto/physics.py

```diff
@@ -685,19 +685,20 @@ def __init__(self, equation, saturation_curve, saturation_variable=None,
         test_v = equation.tests[self.Vv_idx]
 
         # Check if saturation is a function, and if so what of.
-        # This works if saturation is function of one of the equation variables.
-        if isinstance(self.saturation, FunctionType):
-            if self.saturation_variable in equation.field_names:
-                self.variable_idx = equation.field_names.index(
-                    self.saturation_variable)
-                Vvar_space = W.sub(self.variable_idx)
-                self.variable = Function(Vvar_space)
-                self.saturation_function = Function(Vvar_space)
+        # The dependency must be one of the equation variables.
+        if isinstance(self.saturation_curve, FunctionType):
+            if saturation_dependency in equation.field_names:
+                self.dependency_idx = equation.field_names.index(
+                    saturation_dependency)
+                dependency_space = W.sub(self.dependency_idx)
+                self.saturation_dependency = Function(dependency_space)
+                # make a Firedrake function to store the saturation function
+                self.saturation_function = Function(dependency_space)
             else:
                 raise NotImplementedError(
                     "Saturation function must be either constant in time or a function of an equation variable")
         else:
-            self.saturation_function = saturation_curve
+            self.saturation_function = self.saturation_curve
```

Comment:
is `self.saturation_curve` ever used? I'm not sure it needs to be stored in `self`.